src/pages/index.js

```diff
@@ -0,0 +1,238 @@
+import "./index.css";
+import Card from "../components/Card.js";
+import FormValidator from "../components/FormValidator.js";
+import Section from "../components/Section.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithForm from "../components/PopupWithForm.js";
+import UserInfo from "../components/UserInfo.js";
+import { settings, elements } from "../utils/constants.js";
+import Api from "../components/Api.js";
+
+/* ---------- Utils Elements ------------ */
+const {
+  profileEditBtn,
+  profileTitleInput,
+  profileEditForm,
+  profileDescriptionInput,
+  placesAddBtn,
+  placeAddForm,
+  avatarUpdateBtn,
+  avatarForm,
+} = elements;
+
+/* ---------- API Var ------------ */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "ca917973-003f-412f-833a-d01c5747f711",
+    "content-type": "application/json",
+  },
+});
+
+/* ---------- Image Click Preview Function ------------ */
+function handleImageClick(imageData) {
+  imagePreviewModal.open(imageData);
+}
+
+/* ---------- Validation ------------ */
+const profileEditValidation = new FormValidator(settings, profileEditForm);
+const addPlaceValidation = new FormValidator(settings, placeAddForm);
+const avatarValidation = new FormValidator(settings, avatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```